### PR TITLE
Fixed computed output shape of matmul op.

### PR DESCRIPTION
**Pull request by BruceDai**
@mingmingtasd PTAL, and there's the same issue for webnn-native: https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn/native/ops/Binary.cpp#L57, please help fix it, thanks.

### CODE REVIEW

**Review by mingmingtasd — COMMENTED** [review]
LGTM. But there is no test case to cover and validate, should we add more?

**Comment by BruceDai** [comment]
Hi @mingmingtasd, I caught this issue by [this new mamul 4D x 3D test](https://github.com/web-platform-tests/wpt/blob/dc9f6a4ac55caaa672c61ee8d44056c63d73b360/webnn/resources/test_data/matmul.json#L165) for WPT.

**Review by mingmingtasd — APPROVED** [review]
Thanks! LGTM.

**Comment by BruceDai** [comment]
@mingmingtasd Updated, please take another look, thanks.

File: src/nn/ops/binary.ts [hunk 1]
```diff
@@ -31,11 +31,11 @@ export abstract class Binary extends SingleOutputOperation {
...
-          outputShape = [];
-        } else if (rankA === 2 && rankB === 1) {
-          outputShape = [a.shape[0], 1];
-        } else if (rankA === 1 && rankB === 2) {
-          outputShape = [1, b.shape[1]];
+          outputShape = []; // scalar
+        } else if (rankA >= 2 && rankB === 1) {
+          outputShape = [...a.shape.slice(0, -1), 1];
+        } else if (rankA === 1 && rankB >= 2) {
+          outputShape = [...b.shape.slice(0, -2), 1, b.shape[1]];
```
Comment (mingmingtasd) [inline src/nn/ops/binary.ts:38]: outputShape = [...b.shape.slice(0, -2), 1, b.shape[rankB-1]];?

Comment (BruceDai) [reply src/nn/ops/binary.ts:38]: Yes, thanks for pointing this error out, I will update it.

Comment (BruceDai) [reply src/nn/ops/binary.ts]: Updated, PTAL, thanks.

**Review by mingmingtasd — APPROVED** [review]
LGTM, thanks!

**Review by mingmingtasd — APPROVED** [review]

**Pull request merged**